src/components/community/comments/index.tsx

```diff
@@ -0,0 +1,130 @@
+import React, { FC, memo, useEffect, useState } from 'react';
+import { compose } from 'redux';
+import FdkLink from '@fellesdatakatalog/link';
+import {
+  useGetThreadByIdQuery,
+  useGetUserQuery,
+  usePostCommentMutation
+} from '../../../api/user-feedback-api/comments';
+import { withAuth, Props as AuthProps } from '../../../providers/auth';
+import SC from './styled';
+import Buttons from '../../buttons/styled';
+import Composer from './components/composer';
+import CommentCard from './components/comment-card';
+import { CommunityPost } from '../../../types';
+import translations from '../../../lib/localization';
+import env from '../../../env';
+
+const { FDK_COMMUNITY_BASE_URI } = env;
+
+interface ExternalProps {
+  entityId: string;
+}
+
+interface Props extends AuthProps, ExternalProps {}
+
+const EntityComments: FC<Props> = ({ entityId, authService }) => {
+  const [postComment] = usePostCommentMutation();
+  const [rootComments, setRootComments] = useState<CommunityPost[]>([]);
+  const [replyComments, setReplyComments] = useState<
+    Record<string, CommunityPost[]>
+  >({});
+  const [newCommentOpen, setNewCommentOpen] = useState(false);
+  const [showAllComments, setShowAllComments] = useState(false);
+
+  const { data: thread } = useGetThreadByIdQuery(entityId);
+  const { data: currentUser } = useGetUserQuery();
+
+  const maxVisibleComments = 3;
+  const croppedComments = showAllComments
+    ? rootComments
+    : rootComments.slice(0, maxVisibleComments);
+  const authenticated =
+    authService.isAuthenticated() && !authService.isTokenExpired();
+  const isLoggedIn = authenticated && currentUser != null;
+
+  useEffect(() => {
+    const roots: CommunityPost[] = [];
+    const replies: Record<string, CommunityPost[]> = {};
+
+    thread?.posts?.slice(1)?.forEach(comment => {
```

Comment:
For lesbarheten er det kanskje en fordel om du kan trekke dette ut i en navngitt funksjon som sier hva du gjør her.